runtime/cudaq/utils/type_traits.h

```diff
@@ -0,0 +1,119 @@
+/****************************************************************-*- C++ -*-****
```

Comment:
My thought was that this would always be included from relative path `#include "cudaq/utils/type_traits.h"`, thus clearly differentiating from the standard header.